pkg/config/resource/name.go

```diff
@@ -19,6 +19,22 @@ import (
 	"strings"
 )
 
+type ConfigType int
+
+const (
+	ServiceEntryConfigType ConfigType = iota
+	WorkloadEntryConfigType
+	PodConfigType
+)
+
+// ConfigKey unique identifies a config object managed by service registry, such as ServiceEntry and WorkloadEntry
+type ConfigKey struct {
+	Kind      ConfigType
+	Name      string
+	Namespace string
+	Address   string
```

Comment:
I am not sure how and where  is this used.

Reply:
Hi @hzxuzhonghu, this `ConfigKey` struct is mainly used in `[serviceInstancesStore](pilot/pkg/serviceregistry/serviceentry/store.go)` which will cache the endpoints and service instances etc.
The original definition of `configKey` only include `Kind`, `Name` and `Namespace` which is not unique identifier for an endpoint with multiple ip addresses, such as in Dual Stack environment. So I submit this PR to add the `Address` field for support it.  And this PR is depended by https://github.com/istio/istio/pull/44115

Reply:
Without looking at that pr, what i am concerned developer may not know which ffields to set in future. I think previously we set all the kind / namespace /name

KindAddress is special

Reply:
Yeah, there is a problem that how can represent an endpoint with multiple ip addresses, for now the IstioEndpoint just only one `Address` field for it.

Reply:
Yeah, there is a problem that how can represent an endpoint with multiple ip addresses, especially for a WorkloadEntry, for now the IstioEndpoint just only one `Address` field for it.